backend/src/apis/snapshot_data.rs

```diff
@@ -0,0 +1,249 @@
+use num_bigint::BigInt;
+use std::collections::HashMap;
+
+use halo2_proofs::{
+    halo2curves::bn256::Fr as Fp,
+    plonk::{keygen_pk, keygen_vk},
+};
+use snark_verifier_sdk::CircuitExt;
+
+use summa_solvency::{
+    circuits::{
+        merkle_sum_tree::MstInclusionCircuit,
+        solvency::SolvencyCircuit,
+        utils::{full_prover, generate_setup_params},
+    },
+    merkle_sum_tree::utils::big_int_to_fp,
+    merkle_sum_tree::{Entry, MerkleSumTree},
+};
+
+use crate::apis::csv_parser::parse_csv_to_assets;
+
+struct SnapshotData<
+    const LEVELS: usize,
+    const L: usize,
+    const N_ASSETS: usize,
+    const N_BYTES: usize,
+    const K: u32,
+> {
+    exchange_id: String,
+    mst: MerkleSumTree<N_ASSETS>,
+    assets: Vec<Asset>,
+    proofs_of_inclusion: HashMap<u64, InclusionProof>,
+    proof_of_solvency: Option<SolvencyProof<N_ASSETS>>,
```

Comment:
Why this is an option and `proofs_of_inclusions` are not?

Reply:
the generation of proofs using the aggregated circuit is quite time-consuming, 
so I was considering executing it in the background in the future. 

Now, we are using a solvency proof for that, I think it's okay to run when initiating this struct.
But, Let's change it later, as I am keen on gauging the duration of execution within the 'new' method through CLI at a later stage.